agent/tcs/client/client.go

```diff
@@ -221,6 +228,87 @@ func (cs *clientServer) metricsToPublishMetricRequests() ([]*ecstcs.PublishMetri
 	return requests, nil
 }
 
+// publishHelathMetrics send the container health information to backend
+func (cs *clientServer) publishHelathMetrics() {
+	if cs.publishTicker == nil {
+		seelog.Debug("Skipping publishing health metrics. Publish ticker is uninitialized")
+		return
+	}
+
+	// Publish metrics immediately after we connect and wait for ticks. This makes
+	// sure that there is no data loss when a scheduled metrics publishing fails
+	// due to a connection reset.
+	err := cs.publishHealthMetricsOnce()
+	if err != nil {
+		seelog.Warnf("Error publishing health metrics: %v", err)
+	}
+	for {
+		select {
+		case <-cs.publishTicker.C:
+			err := cs.publishHealthMetricsOnce()
+			if err != nil {
+				seelog.Warnf("Error publishing health metrics: %v", err)
+			}
+		case <-cs.endPublish:
+			return
+		}
+	}
+}
+
+// publishHealthMetricsOnce is invoked by the ticker to periodically publish metrics to backend.
+func (cs *clientServer) publishHealthMetricsOnce() error {
+	// Get the list of health request to send to backend.
+	requests, err := cs.createPublishHealthRequests()
+	if err != nil {
+		return err
+	}
+
+	// Make the publish metrics request to the backend.
+	for _, request := range requests {
+		err = cs.MakeRequest(request)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// createPublishHealthRequests creates the requests to publish container health
+func (cs *clientServer) createPublishHealthRequests() ([]*ecstcs.PublishHealthRequest, error) {
+	metadata, taskHealthMetrics, err := cs.statsEngine.GetTaskHealthMetrics()
+	if err != nil {
+		return nil, err
+	}
+
+	if metadata == nil || taskHealthMetrics == nil {
+		seelog.Debug("No container health metrics to report")
+		return nil, nil
+	}
+
+	var requests []*ecstcs.PublishHealthRequest
+	var taskHealths []*ecstcs.TaskHealth
+	numOfTasks := len(taskHealthMetrics)
+	for i, taskHealth := range taskHealthMetrics {
+		taskHealths = append(taskHealths, taskHealth)
+		// create a request if the number of task reaches the maximum page size
+		if (i+1)%tasksInMessage == 0 {
```

Comment:
Please define a new const for `tasksInHealthMessage` and rename `tasksInMessage` to `tasksInMetricsMessage`. Also add a TODO for determining the proper value for `tasksInHealthMessage`. Because of changes in payload structure, we'd have differing values for these I think